mmhuman3d/models/losses/prior_loss.py

```diff
@@ -65,6 +67,99 @@ def forward(self,
         return shape_prior_loss
 
 
+@LOSSES.register_module()
+class LimbLengthLoss(nn.Module):
+    """Limb length loss for body shape parameters. Note that it should take
+    keypoints3d as input, as limb length computed from keypoints2d varies with
+    camera.
+
+    Args:
+        reduction (str, optional): The method that reduces the loss to a
+            scalar. Options are "none", "mean" and "sum".
+        loss_weight (float, optional): The weight of the loss. Defaults to 1.0
+    """
+
+    def __init__(self,
+                 limb_convention,
+                 reduction='mean',
+                 loss_weight=1.0,
+                 eps=1e-4):
+        super().__init__()
+        assert reduction in (None, 'none', 'mean', 'sum')
+        self.reduction = reduction
+        self.loss_weight = loss_weight
+        self.eps = eps
+        self.limb_conf = None
+        self.limb_length = None
+        limb_idxs, _ = search_limbs(data_source=limb_convention)
+        limb_idxs = sorted(limb_idxs['body'])
+        self.limb_idxs = np.array(
+            list(x for x, _ in itertools.groupby(limb_idxs)))
+
+    def _get_limb_vec(self, keypoints3d):
+        kp_src = keypoints3d[:, self.limb_idxs[:, 0], :3]
+        kp_dst = keypoints3d[:, self.limb_idxs[:, 1], :3]
+        limb_vec = (kp_dst - kp_src).detach()
+        return limb_vec
+
+    def _compute_normed_limb(self, keypoints3d):
+        limb_vec = self._get_limb_vec(keypoints3d)
+        limb_length = torch.norm(limb_vec, dim=2, keepdim=True)
+        limb_vec_normed = limb_vec / (limb_length + self.eps)
+        return limb_vec, limb_vec_normed, limb_length
+
+    def forward(self,
+                keypoints3d_pred,
+                keypoints3d_target,
+                keypoints3d_pred_conf=None,
+                keypoints3d_target_conf=None,
+                loss_weight_override=None,
+                reduction_override=None):
+        """Forward function of LimbLengthLoss.
+
+        Args:
+            betas (torch.Tensor): The body shape parameters
+            loss_weight_override (float, optional): The weight of loss used to
+                override the original weight of loss
+            reduction_override (str, optional): The reduction method used to
+                override the original reduction method of the loss.
+                Defaults to None
+        Returns:
+            torch.Tensor: The calculated loss
```

Comment:
The `Args` does not cover all input parameters. The shape of parameters would be especially important to help users understand the loss.